salt/orch/removal.sls

```diff
@@ -149,12 +147,20 @@
 {%- endif %} {# target in minions #}
 
 {#- other consistency checks... #}
+
+{#- consistency check: if there is a replacement, it must have some (new) role(s) #}
 {%- if replacement %}
-  {#- consistency check: if there is a replacement, it must have some (new) role(s) #}
   {%- if not replacement_roles %}
     {%- do salt.caasp_log.abort('CaaS: %s cannot be removed: too few etcd members, and no replacement found', target) %}
   {%- endif %}
-{%- endif %} {# replacement #}
+{%- endif %}
+
+{#- consistency check: if some of the previous checks set replacement to an empty string, clear replacement_roles #}
+{%- if not replacement %}
```

Comment:
Why not a simple `{% else %}` instead of `{% endif %} \n {% if not replacement %}` ?

Reply:
The idea was to keep the consistency checks separate, but I'll change it.